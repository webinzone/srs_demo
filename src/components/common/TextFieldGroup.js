import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";

const TextFieldGroup = ({
  name,
  placeholder,
  value,
  error,
  info,
  type,
  onChange,
  disabled
}) => {
  return (
    <div className="form-group">
      <input
        type={type}
        className={classnames("form-control form-control-lg", {
          //Used classnames package for conditional class setup
          "is-invalid": error //Bootstrap class will be changed to is-invalid if error is present which we are passing in the component
        })}
        placeholder={placeholder}
        name={name}
        value={value}
        onChange={onChange}
        disabled={disabled}
      />
      {info && <small className="form-text text-muted">{info}</small>}
      {error && <div className="invalid-feedback">{error}</div>}
    </div>
  );
};

TextFieldGroup.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  info: PropTypes.string,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
  disabled: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

TextFieldGroup.defaultProps = {
  type: "text",
  name: "",
  value: "",
  onChange: () => {}
};

export default TextFieldGroup;
