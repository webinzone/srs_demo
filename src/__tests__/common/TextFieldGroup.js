//As this is a simple non interactive component,
//testing is done to check basic component rendering
import React from "react";
import { shallow } from "enzyme";

import TextFieldGroup from "../../components/common/TextFieldGroup";

describe("<TextFieldGroup/>", () => {
  it("renders without crashing", () => {
    //Shallow isolates and only renders single component, not parent. So that we get pure unit testing
    const component = shallow(<TextFieldGroup />);

    //This function checks if the snapshot matches the rendered component
    expect(component).toMatchSnapshot();
  });
});
