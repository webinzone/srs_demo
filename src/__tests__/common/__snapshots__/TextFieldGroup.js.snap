// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextFieldGroup/> renders without crashing 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control form-control-lg"
    name=""
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
